### PR TITLE
[FLINK-26332]Move the operator env to common utils

### PR DESCRIPTION
**Pull request by Aitozi**
Move the scattered environments variables to the `EnvUtils` and make it clear whether the variable is required

### CODE REVIEW

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public static final String ENV_OPERATOR_NAMESPACE = "OPERATOR_NAMESPACE";
+
+    /**
+     * Get the value provided by environments.
+     *
+     * @param key the target key
+     * @param required whether to verify the key is exists.
+     * @return the value or default value provided by environments.
+     */
+    public static String get(String key, boolean required) {
```
Comment (morhidi) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java:42]: We could probably use two separate methods get() and getRequired() here.

Comment (Aitozi) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java:52]: I added separate `get` method without `required` parameter

Comment (wangyang0918) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java:52]: nit: I also lean to `getRequired()`.

**Review by wangyang0918 — APPROVED** [review]

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java (hunk 1 above)
Comment (Aitozi) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java]: updated

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    /**
+     * Get the value provided by environments.
+     *
+     * @param key the target key
+     * @param required whether to verify the key is exists.
+     * @return the value provided by environments.
+     */
+    public static String getRequired(String key, boolean required) {
```
Comment (wangyang0918) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java:52]: Hmm. We will not need the second argument `required` now. Right?

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        String value = System.getenv().get(key);
+        if (value == null && required) {
```
Comment (wangyang0918) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java:54]: Do we also need to check the empty string here?

File: flink-kubernetes-webhook/src/main/java/org/apache/flink/kubernetes/operator/admission/FlinkOperatorWebhook.java [hunk 4]
```diff
@@ -117,26 +111,19 @@ protected void initChannel(SocketChannel ch) {
...
-        if (StringUtils.isAnyEmpty(keystorePassword, keystoreType)) {
-            throw new RuntimeException(
-                    WEBHOOK_KEYSTORE_PASSWORD
-                            + " and "
-                            + WEBHOOK_KEYSTORE_TYPE
-                            + " must no be empty");
-        }
-
+        String keystorePassword =
+                EnvUtils.getRequired(EnvUtils.ENV_WEBHOOK_KEYSTORE_PASSWORD, true);
```
Comment (wangyang0918) [inline flink-kubernetes-webhook/src/main/java/org/apache/flink/kubernetes/operator/admission/FlinkOperatorWebhook.java:125]: `getRequired` does not check the empty string.

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java (hunk 2 above)
Comment (gyfora) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EnvUtils.java:52]: getRequired should not have a boolean parameter, it is always true (its in the method name)

**Review by gyfora — CHANGES_REQUESTED** [review]

**Comment by gyfora** [comment]
I added a minor comment, also please rebase on top of your config changes and then it should be good to go :)

**Comment by Aitozi** [comment]
I have rebased the master and addressed the comments, Please take a look again, thanks cc @gyfora @wangyang0918

**Comment by wangyang0918** [comment]
LGTM. +1 for merging.

**Pull request merged**